flo-workflow/src/main/java/com/spotify/flo/EvalContext.java

```diff
@@ -79,21 +79,21 @@
    * method, one can intercept the evaluation flow just at the moment between inputs being ready
    * and when the user supplied function for task processing is being invoked.
    *
-   * <p>The default implementation will simply invoke the function immediately.
+   * <p>The default implementation will simply invoke the function immediately inside a
+   * {@link Value} created by {@link #value(Fn)}.
    *
    * @param taskId     The id of the task being invoked
    * @param processFn  A lazily evaluated handle to the process function
    * @param <T>        The task value type
-   * @return The value of the process function invocation
+   * @return The (deferred) value of the process function invocation
    */
-  default <T> Value<T> invokeProcessFn(TaskId taskId, Fn<Value<T>> processFn) {
-    return processFn.get();
+  default <T> Value<T> invokeProcessFn(TaskId taskId, Fn<T> processFn) {
+    return value(processFn);
   }
 
   /**
-   * When called from within any of the functions passed to {@code processWithContext}, this
-   * method will return the {@link Task} currently being processed. Otherwise an empty value
-   * will be returned.
+   * When called during the evaluation of a task, this method will return the {@link Task}
```

Comment:
If we remove `processWithContext` and thus user code has no way to get at the `EvalContext`, what would call this method? Does it make sense to keep it?

Reply:
I noticed that `currentTask()` is actually used from the builder code as part of the `context()` implementation. So it has to stay.